code/mob/living/silicon/robot.dm

```diff
@@ -1493,8 +1493,8 @@
 
 		if (available_actions.len)
 			available_actions.Insert(1, "Cancel")
-			var/action = input("What do you want to do?", "Cyborg Maintenance") as null|anything in available_actions
-			if (!action)
+			var/action = tgui_input_list(user, "What do you want to do?", "Cyborg Maintenance", available_actions)
+			if (!action || available_actions == "Cancel")
```

Comment:
Shouldn't this be `action == "Cancel"` instead?

Reply:
Oops, yup